pkg/driverbuilder/builder/ubuntu.go

```diff
@@ -59,7 +59,9 @@ func (v *ubuntu) TemplateData(c Config, kr kernelrelease.KernelRelease, urls []s
 	if flavor == "hwe" {
 		headersPattern = "linux-headers*generic"
 	} else {
-		headersPattern = fmt.Sprintf("linux-headers*%s", flavor)
+		// some flavors (ex: lowlatency-hwe) only contain the first part of the flavor in the directory extracted from the .deb
+		// splitting a flavor without a "-" should just return the original flavor back	
+		headersPattern = fmt.Sprintf("linux-headers*%s", strings.Split(flavor, "-")[0])
```

Comment:
Thanks for this MR! And welcome to Falco =)

IIUC, this assumes that the flavor will contain at most one dash ("-")?

Reply:
I guess it assumes that when there are dashes, only the first part is actually interesting.
A side note: wouldn't it be better if we were able to return a list of headersPattern and in the template script cycle through all of them and stop at the first matching one?
Just an idea...

Reply:
You're right - the idea was that only the first part of the flavor is interesting. Take this example here (which is driverkit building from the `test/` folder - an `ubuntu-generic`):
```
DEBU .+ cd /tmp/kernel-download/usr/src/
DEBU + ls -altr
DEBU �total 16
DEBU drwxr-xr-x 24 root root 4096 Oct  5  2020 linux-headers-5.4.0-51
DEBU drwxr-xr-x  4 root root 4096 Oct  5  2020 ..
DEBU drwxr-xr-x  4 root root 4096 Oct  5  2020 .
DEBU drwxr-xr-x  7 root root 4096 Oct 19 21:07 linux-headers-5.4.0-51-generic
DEBU =++ find . -type d -name 'linux-headers*generic'
```
There are 2 folders in `/tmp/kernel-download/usr/src`:
1. `linux-headers-5.4.0-51`
2. `linux-headers-5.4.0-51-generic`

My first thought was just to do `linux-headers*`, but that finds the incorrect folder. We need the script to find the `linux-headers*generic`.

In some cases though, the flavors for Ubuntu get weird, specifically `hwe` kernels. `hwe` kernels will be `hwe` in the URL path, but when extracted from the `.deb` show up as `-generic` lol. That's the reason for the first case:
```golang
	headersPattern := ""
	if flavor == "hwe" {
		headersPattern = "linux-headers*generic"
	} 
```

However, in other cases, such as `lowlatency-hwe` kernels, the directory extracted from the `.deb` just chops off the `-hwe` on the end. That produces errors in the current script:
```
DEBU #+ cd /tmp/kernel-download/usr/src/
DEBU 7++ find . -type d -name 'linux-headers*lowlatency-hwe'
DEBU "++ head -n 1
DEBU ++ xargs readlink -f
DEBU Freadlink: missing operand
```
`'linux-headers*lowlatency-hwe'` won't exist...

So my idea was to just chop off the first part of the flavor, regardless of `how-long-the-flavor-might-be` and search for `linux-headers*<first-part-of-flavor>`. That gets us enough of the flavor string to not match the `linux-headers*` directory, but still match the flavor directory.

Another way we could do this is just check for `lowlatency-hwe` kernels, or maybe just check for `-hwe` and handle it specifically.. I was hoping to avoid one off checks though and go for something generic.

We could also try to move this checking to the script 🤔 I'm unsure I like the idea of trying to list all the flavors possible and iterate over them... What we really need is to just check if the directory extracted has _a_ flavor at all in its name.

Let me know what you think we should do though. I'm not convinced what I did is perfect either haha.

Reply:
To answer @dwindsor's original question: it assumes that the only interesting part of the `flavor` when it comes to that `find` command in the script is the first word of the `flavor`.

So even if the flavor is `apple-orange-banana`, we only care about `apple` because the 2 directories extracted from the `.deb` should look like:
1. `linux-headers-<kernel-release>`
2. `linux-headers-<kernel-release>-apple-orange-banana`

Reply:
I think your solution is good @EXONER4TED! Only time will tell :)

Reply:
Agreed, we're at the mercy of the vendor for how they'll change their naming schemes.